homeassistant/components/climate/homematic.py

```diff
@@ -122,10 +130,12 @@ def max_temp(self):
     def _init_data_struct(self):
         """Generate a data dict (self._data) from the Homematic metadata."""
         # Add state to data dict
-        self._data.update({"CONTROL_MODE": STATE_UNKNOWN,
-                           "SET_TEMPERATURE": STATE_UNKNOWN,
-                           "ACTUAL_TEMPERATURE": STATE_UNKNOWN})
+        self._state = list(self._hmdevice.WRITENODE.keys())[0]
```

Comment:
Either be more efficient with `next(iter(self._hmdevice.WRITENODE.keys()))` or also use changes of PR mentioned above and reduce do `self._hmdevice._set_temp`.